### PR TITLE
feat: add support to validating webhooks

### PR DESCRIPTION
**Pull request by JorTurFer**
Signed-off-by: Jorge Turrado <jorge_turrado@hotmail.es>

To pass CI checks here, the feature needs to be merged in KEDA because the new image is required for CI checks here

### Checklist

- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/charts/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] README is updated with new configuration values *(if applicable)*
- [x] A PR is opened to update KEDA core ([repo](https://github.com/kedacore/keda)) *(if applicable, ie. when deployment manifests are modified)*

Fixes https://github.com/kedacore/keda/issues/3755
Fixes https://github.com/kedacore/keda/issues/3087
Fixes https://github.com/kedacore/keda/issues/3083
Fixes https://github.com/kedacore/keda/issues/3088

### CODE REVIEW

File: keda/templates/14-keda-deployment.yaml [hunk 1]
```diff
@@ -130,14 +139,19 @@ spec:
           {{- end }}
           {{- if .Values.volumes.keda.extraVolumeMounts }}
           {{- toYaml .Values.volumes.keda.extraVolumeMounts | nindent 12 }}
-          {{- end }}
+          {{- end }}
```
Comment (zroubalik) [inline keda/templates/14-keda-deployment.yaml:142]: remove the whitespace

File: keda/templates/33-webhooks-servicemonitor.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    relabelings:
+    {{ toYaml . | indent 6 }}
+    {{- end }}
+  namespaceSelector:
+    matchNames:
+    - {{ .Release.Namespace }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ .Values.webhooks.name }}
+{{- end }}
```
Comment (zroubalik) [inline keda/templates/33-webhooks-servicemonitor.yaml:51]: add new line pls

**Review by zroubalik — COMMENTED** [review]
looking good, just nits

File: keda/templates/33-webhooks-servicemonitor.yaml (hunk 2 above)
Comment (JorTurFer) [reply keda/templates/33-webhooks-servicemonitor.yaml]: done

File: keda/templates/14-keda-deployment.yaml (hunk 1 above)
Comment (JorTurFer) [reply keda/templates/14-keda-deployment.yaml:142]: done

**Review by zroubalik — APPROVED** [review]

**Comment by JorTurFer** [comment]
f**k! I thought I was clicking in auto-merge but I this repo doesn't have it because there isn't required checks, I hope all will pass, if not I'll solve the issues in other PR

File: keda/values.yaml [hunk 3]
```diff
@@ -52,6 +56,23 @@ metricsServer:
     #         - keda-operator-metrics-apiserver
     #     topologyKey: "kubernetes.io/hostname"
 
+webhooks:
+  enabled: true # This value will be removed in keda v2.12
```
[the remark this answers is not in the document]
Comment (JorTurFer) [reply keda/values.yaml:60]: Because we talked about making this not optional in KEDA, Am I wrong?

[the remark this answers is not in the document]
Comment (JorTurFer) [reply keda/values.yaml:60]: I have to solve the issues I generated with the early merge, I can remove the comment if you want

**Pull request merged**